fsclient/client.go

```diff
@@ -0,0 +1,115 @@
+package fsclient
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"log"
+	"mime/multipart"
+	"net/http"
+	url2 "net/url"
+)
+
+type FileServerClient struct {
+	adr string
+	username string
+	password string
+}
+
+func NewFileServerClient(adr string, username string, password string) *FileServerClient{
+	return &FileServerClient{
+		adr: adr,
+		username: username,
+		password: password,
+	}
+}
+
+func (fsc *FileServerClient) UploadFile (url string, reader *io.Reader, filename string) error {
```

Comment:
accept an io.Reader and not a pointer to it (interface values are always pointers)